specification/connector/_types/Connector.ts

```diff
@@ -0,0 +1,231 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+import { Id } from '@_types/common'
+import { Dictionary } from '@spec_utils/Dictionary'
+import { UserDefinedValue } from '@spec_utils/UserDefinedValue'
+
+interface SelectOption {
+  label: string
+  value: string
+}
+
+interface Dependency {
+  field: string
+  value: string | number | boolean | null
+}
+
+enum DisplayType {
+  TEXTBOX = 'textbox',
+  TEXTAREA = 'textarea',
+  NUMERIC = 'numeric',
+  TOGGLE = 'toggle',
+  DROPDOWN = 'dropdown'
+}
+
+enum ConnectorFieldType {
+  STRING = 'str',
+  INTEGER = 'int',
+  LIST = 'list',
+  BOOLEAN = 'bool'
+}
+
+interface Validation {
+  constraint: string | number
```

Comment:
It's limited to strings and numbers. It's represented as an `Object` type in ES, though the [parser](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/configuration/ConfigurationValidation.java#L65) validates that it's either of numeric or string type

Reply:
Oh, as I understand, `type` is an [enumeration of the possible constraints](https://github.com/elastic/elasticsearch/blob/main/x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/connector/configuration/ConfigurationValidationType.java) that precisely defines what should be in the `constraint` field.

So this can be represented in a fully-typed way as follows using an union defining "[internally tagged variants](https://github.com/elastic/elasticsearch-specification/blob/main/docs/modeling-guide.md#internal)":

```ts
/** @variants internal tag='type' */
export type Validation = LessThanValidation | GreaterThanValidation | ListTypeValidation | IncludedInValidation | RegexValidation | UnsetValidation

export interface LessThanValidation {
  type: 'less_than'
  constraint: double // we want to avoid 'number'
}

export interface GreaterThanValidation {
  type : 'greater_than'
  constraint: double
}

export interface ListTypeValidation {
  type: 'list_type'
  constraint: string
}

export interface IncludedInValidation {
  type: 'included_in'
  constraint: string // is it really a string?
}

export interface RegexValidation {
  type: 'regex'
  constraint: string
}

export interface UnsetValidation {
  type: 'unset'
  // no constraint ?
}
```

Reply:
Ack, thank you for explanation! Will include this into the type definitions